### PR TITLE
CODE: Add Linters configs. Just added linters to md, link check, spel…

### PR DESCRIPTION
**Pull request by madossan01**
## Overview

Added all basic linters related to powershell code standard, markdown, spelling and link checkers.

Jscp was removed from linting for now, but it can be added back in future PR.

## Details
[Move devcontainer (Voyager, AIO repo Katrien's repo) and other validations (linters) to our new repo#412](https://github.com/cse-labs/distributed-az-edge-framework/issues/412)

## Checklist

[READY TO PR? Use the check-list below to ensure your branch is ready for PR.
If the item is not applicable, leave it unchecked.]

- [x] I have checked my changes follow the repo structure and land in the appropriate folder.
- [ ] I have run design discussion before pushing this PR (if code PR).
- [x] I have updated the applicable documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added integration tests to check if my changes are integrating accordingly.
- [ ] All new and existing tests passed.
- [ ] I ran the lint checks which produced no new errors nor warnings for my changes.
- [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change.
- [x] My changes build properly.
- [ ] My changes execute properly.

### CODE REVIEW

File: docs/workflows/linters.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Linters
```
Comment (olhakonst17) [inline docs/workflows/linters.md:1]: We might be able to delete a .gitkeep file now when we have some files in docs directory.

File: .cspell.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (olhakonst17) [inline .cspell.json:1]: Just curious if it's necessary for this linting related files to be in the root directory? Or maybe we can group them all within /linting directory under .github?

File: README.md [hunk 3]
```diff
@@ -2,24 +2,24 @@
...
 
-The repository is divided into the following key directories:
+We organize this repository into the following key directories:
 
 - **Docs** directory - This directory contains any supporting documentation, such as architecture and solutions walkthrough.
 
 - **infra** directory - This directory contains infrastructure implementation, including infrastructure provisioning scripts under "provisioning" subfolder and workload deployment scripts under "deployment" subfolder.
 
-- **src** directory - This directory contains business logic implementation. It includes two subfolders: "low-code" and "pro-code" that contain logic implemented with AIO data processor pipelines low-code approach and with C# pro-code approach accordingly.
+- **src** directory - This directory contains business logic implementation. It includes two sub-folders: "low-code" and "pro-code" that contain logic implemented with AIO data processor pipelines low-code approach and with C# pro-code approach accordingly.
```
Comment (olhakonst17) [inline README.md:11]: I've noticed this change - when writing this I wasn't sure how it's writing correctly and double checked before pushing this, and I've found that both ways seem to be correct. And actually, copilot mentioned that subfolder is more commonly used in technical writing.

**Review by olhakonst17 — APPROVED** [review]
Thank you for adding linters!

Left a few nit comments.

File: .cspell.json (hunk 2 above)
Comment (madossan01) [reply .cspell.json:1]: As far I can tell, Megalinter looks into root for these files. Even in megalinter repo it is located in the root folder. I can investigate further if there is a way to force it to look into another folder.

File: README.md (hunk 3 above)
Comment (madossan01) [reply README.md:11]: Yes, both are grammatically correct, but 'sub-folders' is the only form recognized automatically by CSPELL. If you prefer to keep it as subfolders, I can roll back the changes and add it to ignore words.

File: docs/workflows/linters.md (hunk 1 above)
Comment (madossan01) [reply docs/workflows/linters.md:1]: Good call out. Thanks.

File: README.md (hunk 3 above)
Comment (olhakonst17) [reply README.md]: Got it, thanks for explaining!

**Pull request merged**